File: week01/pset1/cash/less/cash.c
// Find out the minimum number of coins for change, coins in denominations of 25, 10, 5, 1 cent

#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars <= 0);

    int cents = round(dollars * 100);

    int count = 0;

    count = cents / 25;
    cents = cents % 25;

    count += cents / 10;
    cents = cents % 10;

    count += cents / 5;
    cents = cents % 5;

    count += cents;

    printf("%i\n", count);
}

File: week06/pset6/cash/cash.py
def main():
    while True:
        try:
            owned = float(input("Change owed: "))
            if owned > 0:
                break
        except:
            continue

    monets = [25, 10, 5, 1]
    cents = round(owned * 100)
    count = 0

    for n in monets:
        count += cents // n
        cents = cents % n

    print(count)


if __name__ == "__main__":
    main()
